pkgs/build-support/zig/build-zig-package/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, zig }:
+
+{ strictDeps ? true
+, nativeBuildInputs ? [ ]
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  buildPhase = attrs.buildPhase or ''
+    runHook preBuild
+    export HOME=$(mktemp -d)
+    zig build -Drelease-safe -Dcpu=baseline $zigBuildFlags
+    runHook postBuild
+  '';
+
+  checkPhase = attrs.checkPhase or ''
+    runHook preCheck
+    zig build test
+    runHook postCheck
+  '';
+
+  installPhase = attrs.installPhase or ''
+    runHook preInstall
+    zig build $zigBuildFlags --prefix $out install
```

Comment:
`zigBuildFlags ` is a string or a list of strings?

Reply:
it is supposed to be a list of strings, but a regular string will work as well

Reply:
how would I use it, cause I am trying to package river.

Reply:
```

Reply:
just do `zigBuildFlags = [ "-Dexample-flag" ];`